File: yt/yt/flow/examples/java/static_table_join/static_table_join/src/main/java/tech/ytsaurus/flow/examples/statictablejoin/StaticTableJoinComputationContext.java
package tech.ytsaurus.flow.examples.statictablejoin;

import java.util.List;

import org.springframework.context.annotation.Configuration;
import tech.ytsaurus.flow.computation.Computation;
import tech.ytsaurus.flow.spring.ComputationProvider;

// [BEGIN computation_context]
@Configuration
public class StaticTableJoinComputationContext implements ComputationProvider {

    @Override
    public List<Computation> getComputations() {
        return List.of(
                Computation.builder()
                        .setComputationId("reference_loader")
                        .setProcessFunction(new ReferenceLoaderFunction())
                        .build(),
                Computation.builder()
                        .setComputationId("enricher")
                        .setProcessFunction(new EnricherFunction())
                        .build()
        );
    }
}
// [END computation_context]

File: yt/yt/flow/examples/kotlin/static_table_join/static_table_join/src/main/kotlin/tech/ytsaurus/flow/examples/statictablejoin/StaticTableJoinComputationContext.kt
package tech.ytsaurus.flow.examples.statictablejoin

import org.springframework.context.annotation.Configuration
import tech.ytsaurus.flow.computation.Computation
import tech.ytsaurus.flow.spring.ComputationProvider

// [BEGIN computation_context]
@Configuration
open class StaticTableJoinComputationContext : ComputationProvider {
    override fun getComputations(): List<Computation> {
        return listOf(
            Computation.builder()
                .setComputationId("reference_loader")
                .setProcessFunction(ReferenceLoader())
                .build(),
            Computation.builder()
                .setComputationId("enricher")
                .setProcessFunction(Enricher())
                .build()
        )
    }
}
// [END computation_context]
